### ISSUE TITLE
Bug: [no-shadow] False positives on enum

### ISSUE THREAD

**Issue by JasonMan34** [typescript-eslint/typescript-eslint#7116]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#sourceType=module&code=MYewdgzgLgBAKgU2jAvDA5AQwEbHQbgChCEwBXAWxgFkBPAUXKoG9CZ4koAaNmAZQSgwAEx4BfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uQAsBDACYcA7uiiJo0DtEjgwAXxDygA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const Test = 'abc';

enum MyEnum {
  Test,
  Second,
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-shadow": "error"
  },
};
```

### Expected Result

No error

### Actual Result

I am getting an error -  `'Test' is already declared in the upper scope on line 1 column 7. 4:3 - 4:7`  
However, I'm not shadowing the `Test` variable, it is the key of an enum.

### Additional Info

If I wrap the enum keys in quotation marks, the error disappears, even though the same code is outputted:

```ts
const Test = 'abc';

enum MyEnum {
  // No error now, even though it's functionally the same
  "Test",
  Second,
}
```

**Comment by Josh-Cena** [comment]
The fact that using a string literal prevents the report is probably a bug, but the report is correct. See https://github.com/typescript-eslint/typescript-eslint/issues/5985 and its many duplicates.

**Comment by bradzacher** [comment]
Yup - it's even called out in the docs:
https://typescript-eslint.io/rules/no-shadow/#why-does-the-rule-report-on-enum-members-that-share-the-same-name-as-a-variable-in-a-parent-scope

The string key case isn't caught because it's SUPER DUPER rare that anyone actually uses them in enums - let alone uses them in enums for keys that don't need quotes.
TBH i'm okay with leaving it as an unhandled case as well, given how so very rare it is.
